test/upgrade/prober/verify.go

```diff
@@ -59,8 +62,10 @@ func (p *prober) Verify() (eventErrs []error, eventsSent int) {
 		p.client.Kube, p.client.T.Logf, system.Namespace()); err != nil {
 		p.log.Warnf("Failed to setup Zipkin tracing. Traces for events won't be available.")
 	} else {
+		lock.Lock()
 		// Required for proper cleanup.
 		zipkin.ZipkinTracingEnabled = true
+		lock.Unlock()
```

Comment:
Don't we need to acquire the lock on read as well?

Maybe worth making a breaking change in pkg and transforming `ZipkinTracingEnabled` a `atomic.Bool` ?

Reply:
Thanks!

Can we track this improvement in an issue?

/lgtm

Reply:
/cherry-pick release-1.9

Reply:
Yes. I will create one and sending a PR in a little while...

Reply:
https://github.com/knative/pkg/issues/2689